girder/girder_large_image/web_client/vue/components/FrameSelector.vue

```diff
@@ -0,0 +1,194 @@
+<script>
+import Vue from 'vue';
+import CompositeChannels from './CompositeChannels.vue';
+import DualInput from './DualInput.vue'
+export default Vue.extend({
+    props: ['imageMetadata', 'frameUpdate'],
+    components: { CompositeChannels, DualInput },
+    data() {
+        return {
+            currentFrame: 0,
+            maxFrame: this.imageMetadata.frames.length - 1,
+            sliderModes: [
+                { id: 0, name: 'Frame' },
+                { id: 1, name: 'Axis' }
+            ],
+            compositeModes: [
+                { id: 0, name: 'Slider' },
+                { id: 1, name: 'Composite' }
+            ],
+            currentModeId: 1,
+            currentChannelNumber: 0,
+            currentChannelCompositeModeId: 0,
+            currentBandCompositeModeId: 1,
+            indices: [],
+            indexInfo: {},
+            compositedFrames: {},
+            activeChannels: [],
+        };
+    },
+    watch: {
+        currentChannelCompositeModeId() {
+            this.updateFrame();
+        }
+    },
+    computed: {
+        nonCompositeIndices() {
+            return this.indices.filter((index) => {
+                return index != 'IndexC' || this.currentChannelCompositeModeId === 0
+            })
+        }
+    },
+    methods: {
+        updateStyle(style) {
+            this.style = style
+            this.updateFrame()
+        },
+        updateFrameByAxes(event) {
+            this.indexInfo[event.index].current = event.frame;
+            this.updateFrame();
+        },
+        updateFrameSimple(newFrame) {
+            Object.keys(this.indexInfo).forEach((key) => {
+                this.indexInfo[key].current = Math.floor(
+                    this.currentFrame / this.indexInfo[key].stride) % this.indexInfo[key].range;
+            });
+            this.frameUpdate(newFrame);
+        },
+        updateChannel(newChannel) {
+            this.currentChannelNumber = newChannel
+            this.updateFrame()
+        },
+        updateFrame() {
+            let frame = 0;
+            _.forEach(this.indices, (index) => {
+                const info = this.indexInfo[index];
+                frame += info.current * info.stride;
+            });
+            const useStyle = this.currentModeId === 1
+                && this.currentChannelCompositeModeId === 1
+            let style = useStyle ? this.style : undefined
+            this.frameUpdate(frame, style);
+        },
+    },
+    mounted() {
+        if (
+            (!this.imageMetadata.channels || this.imageMetadata.channelmap)
```

Comment:
I think this was supposed to be `|| !this.imageMetadata.channelmap)`

Reply:
Thanks for catching that. Changed in [81b50f0](https://github.com/girder/large_image/pull/1086/commits/81b50f0462293ec3b4e530bf95056dffe608ca04)

Reply:
Changed in [81b50f0](https://github.com/girder/large_image/pull/1086/commits/81b50f0462293ec3b4e530bf95056dffe608ca04)